src/components/Sticky/sidebar.js

```diff
@@ -0,0 +1,333 @@
+    this.world = {
+      forced: options.forced || true,
+      tearing: options.tearing || false,
+      tolerance: options.tolerance || 110,
+      lastScrollTop: 0,
+      direction: 'DOWN',
+      boundaries: {}
+    }
+  }
+
+  init () {
+    if (!this.workspace) {
+      this.workspace = this.options.workspace ? document.querySelector(this.options.workspace) : window
+    }
+    if (!this.container) {
+      this.container = document.querySelector(this.containerSelector)
+    }
+    if (!this.content) {
+      this.content = document.querySelector(this.contentSelector)
+    }
+  }
+
+  on () {
+    this.init()
+    this.workspace.addEventListener('scroll', this.onScroll.bind(this))
+  }
+
+  off () {
+    this.init()
+    this.workspace.removeEventListener('scroll', this.onScroll.bind(this))
+  }
+
+  onScroll () {
+    this.init()
+    if (canBeSticky(this.container, this.content, this.options)) {
+      this.last = window.scrollY
+      if (!this.ticking) {
+        this.ticking = true
+
+        // Direction
+        let sd = scrollDirection(this.workspace, this.world.lastScrollTop)
+        this.world.lastScrollTop = sd.scrollTop
+        this.world.direction = sd.direction
+
+        // Boundaries
+        this.world.boundaries = boundaries(this.container, this.content)
+
+        // Positioning
+        applyPositioning(this.container, this.content, this.world, this.options)
+        this.ticking = false
+      }
+    } else {
+      this.content.style.position = ''
+      this.content.style.top = ''
+      removeClass(this.content, 'sidebar-scrolling-down')
+      removeClass(this.content, 'sidebar-bottom-fixed')
+      removeClass(this.content, 'sidebar-scrolling-up')
+      removeClass(this.content, 'sidebar-top-fixed')
+      removeClass(this.content, 'sidebar-scrolling-up')
+    }
+
+    if (this.options.callback) {
+      this.options.callback(this.content, this.world)
+    }
+
+    debug(this.options.log,
+          'world lastScrollTop: ' + this.world.lastScrollTop + '<br/>' +
+          'world direction: ' + this.world.direction + '<br/>' +
+          'world boundaries containerTop: ' + this.world.boundaries.containerTop + '<br/>' +
+          'world boundaries containerHeight: ' + this.world.boundaries.containerHeight + '<br/>' +
+          'world boundaries containerBottom: ' + this.world.boundaries.containerBottom + '<br/>' +
+          'world boundaries contentHeight: ' + this.world.boundaries.contentHeight + '<br/>' +
+          'world boundaries windowHeight: ' + this.world.boundaries.windowHeight + '<br/>' +
+          'positioning applied' + '<br/>' +
+          'content position: ' + this.content.style.position + '<br/>' +
+          'content top: ' + this.content.style.top + '<br/>' +
+          'content class: ' + this.content.className)
+  }
+}
+
+/**
+ * Simple debugging function.
+ */
+const debug = (selector, str) => {
+  let c = document.querySelector(selector)
+  if (c) {
+    c.innerHTML = str
+  }
+}
```

Comment:
I imagine this was very interesting to get working, so I understand the purpose of this `debug` function.  Do y'all want to leave it for now until this sidebar kind of proves itself out a little first?  In some future world, it should probably be removed, but having dealt with this kind of code before, I (a) appreciate how nicely you've done the debug output and (b) understand why it might be valuable to leave it in for a while.

Reply:
This is still a work in progress with https://github.com/truetandem/e-QIP-prototype/issues/2161 so we left it in there, but it should definitely be removed once that is finished. Will put a comment in associated issue!